templates/web/pages/product_edit/product_edit_form_packagings.tt.html

```diff
@@ -0,0 +1,182 @@
+                    <input id="[% packaging_id %]_number_of_units" name="[% packaging_id %]_number_of_units" type="text" value="[% packaging.number_of_units %]">
+                </div>
+            </div>
+        </div>
+
+        [% FOREACH property IN ["shape", "material", "recycling"] %]
+
+            [% IF property == "shape" %]
+                [% SET taxonomy = "packaging_shapes" %]
+            [% ELSIF property == "material" %]
+                [% SET taxonomy = "packaging_materials" %]
+            [% ELSIF property == "recycling" %]
+                [% SET taxonomy = "packaging_recycling" %]
+            [% END %]
+
+            [% SET select2_id = packaging_id _ "_$property" %]
+
+        <div class="small-12 large-3 columns">
+            <label class="hide-for-large-up" for="[% select2_id %]">[% lang("packaging_$property") %]</label>
+            <select id="[% select2_id %]" name="[% select2_id %]" class="form-control [% taxonomy %]_select2">
+                <option></option>
+            </select>
+        </div>
+
+    [% IF packaging_max > 0 %]
+<initjs>
+\$("#[% select2_id %]").select2({
+    placeholder: "[% lang("packaging_$property") %]",
+    data: [% taxonomy %]_select2_options,
+    allowClear:true,
+    tags: true
+});
+
+[% IF packaging.$property.defined %]
+// Set the value, creating a new option if necessary
+var value = "[% display_taxonomy_tag(taxonomy, packaging.$property) %]";
+if (\$("#[% select2_id %]").find("option[value='" + value.replace(/'/g, "\\'") + "']").length) {
+    \$("#[% select2_id %]").val(value).trigger('change');
+} else { 
+    // Create a DOM Option and pre-select by default
+    var newOption = new Option(value, value, true, true);
+    // Append it to the select
+    \$("#[% select2_id %]").append(newOption).trigger('change');
+} 
+[% END %]
+
+\$("#[% packaging_id %]_delete").click(function(event) {
+    event.stopPropagation();
+    event.preventDefault();
+    \$("#[% packaging_id %]_row").remove();
+});
+</initjs>
+    [% END %]
+                
+        [% END %]
+
+        <div class="small-12 large-1 columns">
+            <label class="hide-for-large-up" for="[% packaging_id %]_weight_measured">[% lang("packaging_weight") %] (g)</label>
+            <input id="[% packaging_id %]_weight_measured" name="[% packaging_id %]_weight_measured" type="text" value="[% packaging.weight_measured %]" %]">
+        </div>
+
+        <div class="small-12 large-1 columns">
+            <label class="hide-for-large-up" for="[% packaging_id %]_quantity_per_unit">[% lang("packaging_quantity_per_unit") %]</label>
+            <input id="[% packaging_id %]_quantity_per_unit" name="[% packaging_id %]_quantity_per_unit" type="text" value="[% packaging.quantity_per_unit %]" %]">
+        </div>        
+
+    </div>
+[% END %]
+
+<input type="hidden" name="packaging_max" value="[% packaging_max %]">
+
+<initjs>
+var packaging_max = [% packaging_max %];
```

Comment:
I prefer to track those kind of long range dependencies as they may be harmful in the future !

```suggestion
// defined by the FOREACH loop above
var packaging_max = [% packaging_max %];
```

But why not instead:

```suggestion
var packaging_max = [% product.packagings.length %];
```
(see [length virtual method](https://metacpan.org/dist/Template-Toolkit/view/lib/Template/Manual/VMethods.pod#length)